lib/Sema/TypeChecker.cpp

```diff
@@ -731,6 +732,7 @@ void swift::performWholeModuleTypeChecking(SourceFile &SF) {
   Ctx.diagnoseObjCMethodConflicts(SF);
   Ctx.diagnoseObjCUnsatisfiedOptReqConflicts(SF);
   Ctx.diagnoseUnintendedObjCMethodOverrides(SF);
+  performTopLevelDeclDiagnostics(SF);
```

Comment:
This section is only for diagnostics that apply across files. `performTopLevelDeclDiagnostics` is something we can just do in the regular `performTypeChecking`, and ideally only on the newly-added decls.